pandas/io/parsers.py

```diff
@@ -3278,6 +3278,8 @@ def _isindex(colspec):
             if is_scalar(colspec):
                 if isinstance(colspec, int) and colspec not in data_dict:
                     colspec = orig_names[colspec]
+                elif colspec not in orig_names:
+                    raise ValueError(f"'{colspec}' is not in list")
```

Comment:
```suggestion
                    raise ValueError(f"Missing column provided to ‘parse_dates’: '{colspec}'")
```

Not tied to the wording but I don’t think “is not in list” is super clear to the end user, so could improve a bit

Reply:
I had missed to check CParser, handled it now.